src/app/shared/services/workshops/app-workshop/app-workshops.service.ts

```diff
@@ -83,14 +89,14 @@ export class AppWorkshopsService {
       params = params.set('OrderByField', Ordering.nearest);
       params = params.set('Size', '100');
       params = params.set('From', '0');
-    } else if (filters.currentPage) {
-      const size: number = PaginationConstants.ITEMS_PER_PAGE_DEFAULT;
-      const from: number = size * (+filters.currentPage.element - 1);
-
-      params = params.set('Size', size.toString());
-      params = params.set('From', from.toString());
     }
 
+    const currentPage = this.store.selectSnapshot(PaginatorState.currentPage) as PaginationElement;
+    const size: number = this.store.selectSnapshot(PaginatorState.workshopsPerPage);
+    const from: number = size * (+currentPage.element - 1);
+    params = params.set('Size', size.toString());
+    params = params.set('From', from.toString());
```

Comment:
can you please add check for isMapView. If it is isMapView, then set 0-100 elements 
![image](https://user-images.githubusercontent.com/27493959/171627582-1d24cfe4-6659-473c-a945-d819dfe3bd62.png)